File: 3685-count-subarrays-of-length-three-with-a-condition/3685-count-subarrays-of-length-three-with-a-condition.go
func countSubarrays(nums []int) int {
    res := 0
    n := len(nums)
    for i := 0; i < n-2; i++ {
        if 2*nums[i]+2*nums[i+2] == nums[i+1] {
            res++
        }
    }
    return res
}

File: 3685-count-subarrays-of-length-three-with-a-condition/3685-count-subarrays-of-length-three-with-a-condition.cpp
class Solution {
public:
    int countSubarrays(vector<int>& nums) {
        int res = 0, n = nums.size();
        for(int i = 0; i < n - 3; i++) {
            if(2 * nums[i] + 2 * nums[i+2] == nums[i+1]) res++;
        }
        return res;
    }
};